server/testdb/data_stores.go

```diff
@@ -76,7 +76,7 @@ func (td *postgresDB) UpdateDataStore(ctx context.Context, dataStore model.DataS
 
 	// keep the same creation date to keep sort order
 	dataStore.CreatedAt = oldDataStore.CreatedAt
-	dataStore.ID = dataStore.Slug()
+	dataStore.ID = oldDataStore.ID
```

Comment:
I removed the auto generated ID when updating a data store because this might lead to inconsistencies. Even with the CLI it was a weird behavior because if you run two updates changing the name of the data store, the second one fails because the first changed the data store id.